lib/msf/core/payload/python/meterpreter_loader.rb

```diff
@@ -142,6 +144,214 @@ def stage_meterpreter(opts={})
+	U4=[0x00000000,0x090d0b0e,0x121a161c,0x1b171d12,0x24342c38,0x2d392736,0x362e3a24,0x3f23312a,0x48685870,0x4165537e,0x5a724e6c,0x537f4562,0x6c5c7448,0x65517f46,0x7e466254,0x774b695a,0x90d0b0e0,0x99ddbbee,0x82caa6fc,0x8bc7adf2,0xb4e49cd8,0xbde997d6,0xa6fe8ac4,0xaff381ca,0xd8b8e890,0xd1b5e39e,0xcaa2fe8c,0xc3aff582,0xfc8cc4a8,0xf581cfa6,0xee96d2b4,0xe79bd9ba,0x3bbb7bdb,0x32b670d5,0x29a16dc7,0x20ac66c9,0x1f8f57e3,0x16825ced,0x0d9541ff,0x04984af1,0x73d323ab,0x7ade28a5,0x61c935b7,0x68c43eb9,0x57e70f93,0x5eea049d,0x45fd198f,0x4cf01281,0xab6bcb3b,0xa266c035,0xb971dd27,0xb07cd629,0x8f5fe703,0x8652ec0d,0x9d45f11f,0x9448fa11,0xe303934b,0xea0e9845,0xf1198557,0xf8148e59,0xc737bf73,0xce3ab47d,0xd52da96f,0xdc20a261,0x766df6ad,0x7f60fda3,0x6477e0b1,0x6d7aebbf,0x5259da95,0x5b54d19b,0x4043cc89,0x494ec787,0x3e05aedd,0x3708a5d3,0x2c1fb8c1,0x2512b3cf,0x1a3182e5,0x133c89eb,0x082b94f9,0x01269ff7,0xe6bd464d,0xefb04d43,0xf4a75051,0xfdaa5b5f,0xc2896a75,0xcb84617b,0xd0937c69,0xd99e7767,0xaed51e3d,0xa7d81533,0xbccf0821,0xb5c2032f,0x8ae13205,0x83ec390b,0x98fb2419,0x91f62f17,0x4dd68d76,0x44db8678,0x5fcc9b6a,0x56c19064,0x69e2a14e,0x60efaa40,0x7bf8b752,0x72f5bc5c,0x05bed506,0x0cb3de08,0x17a4c31a,0x1ea9c814,0x218af93e,0x2887f230,0x3390ef22,0x3a9de42c,0xdd063d96,0xd40b3698,0xcf1c2b8a,0xc6112084,0xf93211ae,0xf03f1aa0,0xeb2807b2,0xe2250cbc,0x956e65e6,0x9c636ee8,0x877473fa,0x8e7978f4,0xb15a49de,0xb85742d0,0xa3405fc2,0xaa4d54cc,0xecdaf741,0xe5d7fc4f,0xfec0e15d,0xf7cdea53,0xc8eedb79,0xc1e3d077,0xdaf4cd65,0xd3f9c66b,0xa4b2af31,0xadbfa43f,0xb6a8b92d,0xbfa5b223,0x80868309,0x898b8807,0x929c9515,0x9b919e1b,0x7c0a47a1,0x75074caf,0x6e1051bd,0x671d5ab3,0x583e6b99,0x51336097,0x4a247d85,0x4329768b,0x34621fd1,0x3d6f14df,0x267809cd,0x2f7502c3,0x105633e9,0x195b38e7,0x024c25f5,0x0b412efb,0xd7618c9a,0xde6c8794,0xc57b9a86,0xcc769188,0xf355a0a2,0xfa58abac,0xe14fb6be,0xe842bdb0,0x9f09d4ea,0x9604dfe4,0x8d13c2f6,0x841ec9f8,0xbb3df8d2,0xb230f3dc,0xa927eece,0xa02ae5c0,0x47b13c7a,0x4ebc3774,0x55ab2a66,0x5ca62168,0x63851042,0x6a881b4c,0x719f065e,0x78920d50,0x0fd9640a,0x06d46f04,0x1dc37216,0x14ce7918,0x2bed4832,0x22e0433c,0x39f75e2e,0x30fa5520,0x9ab701ec,0x93ba0ae2,0x88ad17f0,0x81a01cfe,0xbe832dd4,0xb78e26da,0xac993bc8,0xa59430c6,0xd2df599c,0xdbd25292,0xc0c54f80,0xc9c8448e,0xf6eb75a4,0xffe67eaa,0xe4f163b8,0xedfc68b6,0x0a67b10c,0x036aba02,0x187da710,0x1170ac1e,0x2e539d34,0x275e963a,0x3c498b28,0x35448026,0x420fe97c,0x4b02e272,0x5015ff60,0x5918f46e,0x663bc544,0x6f36ce4a,0x7421d358,0x7d2cd856,0xa10c7a37,0xa8017139,0xb3166c2b,0xba1b6725,0x8538560f,0x8c355d01,0x97224013,0x9e2f4b1d,0xe9642247,0xe0692949,0xfb7e345b,0xf2733f55,0xcd500e7f,0xc45d0571,0xdf4a1863,0xd647136d,0x31dccad7,0x38d1c1d9,0x23c6dccb,0x2acbd7c5,0x15e8e6ef,0x1ce5ede1,0x07f2f0f3,0x0efffbfd,0x79b492a7,0x70b999a9,0x6bae84bb,0x62a38fb5,0x5d80be9f,0x548db591,0x4f9aa883,0x4697a38d]
```

Comment:
If you're concerned about space, you could initialize these static arrays from binary data using the array module like:

```
	import base64 as b
	import array from array
	S  = array('B', b.b64decode('Y3x3e/Jrb8UwAWcr/terdsqCyX36WUfwrdSir5ykcsC3/ZMmNj/3zDSl5fFx2DEVBMcjwxiWBZoHEoDi6yeydQmDLBobblqgUjvWsynjL4RT0QDtIPyxW2rLvjlKTFjP0O+q+0NNM4VF+QJ/UDyfqFGjQI+SnTj1vLbaIRD/89LNDBPsX5dEF8Snfj1kXRlzYIFP3CIqkIhG7rgU3l4L2+AyOgpJBiRcwtOsYpGV5HnnyDdtjdVOqWxW9Opleq4IunglLhymtMbo3XQfS72LinA+tWZIA/YOYTVXuYbBHZ7h+JgRadmOlJseh+nOVSjfjKGJDb/mQmhBmS0PsFS7Fg=='))
	Si = array('B', b.b64decode('Uglq1TA2pTi/QKOegfPX+3zjOYKbL/+HNI5DRMTe6ctUe5QypsIjPe5MlQtC+sNOCC6hZijZJLJ2W6JJbYvRJXL49mSGaJgW1KRczF1ltpJscEhQ/e252l4VRlenjZ2EkNirAIy80wr35FgFuLNFBtAsHo/KPw8Cwa+9AwETims6kRFBT2fc6pfyz87wtOZzlqx0IuetNYXi+TfoHHXfbkfxGnEdKcWJb7diDqoYvhv8Vj5LxtJ5IJrbwP54zVr0H92oM4gHxzGxEhBZJ4DsX2BRf6kZtUoNLeV6n5PJnO+g4DtNrir1sMjruzyDU5lhFysEfrp31ibhaRRjVSEMfQ=='))
	T1 = array('I', b.b64decode('pWNjxoR8fPiZd3fujXt79g3y8v+9a2vWsW9v3lTFxZFQMDBgAwEBAqlnZ859KytWGf7+52LX17Xmq6tNmnZ27EXKyo+dgoIfQMnJiYd9ffoV+vrv61lZsslHR44L8PD77K2tQWfU1LP9oqJf6q+vRb+cnCP3pKRTlnJy5FvAwJvCt7d1HP394a6Tkz1qJiZMWjY2bEE/P34C9/f1T8zMg1w0NGj0paVRNOXl0Qjx8fmTcXHic9jYq1MxMWI/FRUqDAQECFLHx5VlIyNGXsPDnSgYGDChlpY3DwUFCrWami8JBwcONhISJJuAgBs94uLfJuvrzWknJ07NsrJ/n3V16hsJCRKeg4MddCwsWC4aGjQtGxs2sm5u3O5aWrT7oKBb9lJSpE07O3Zh1ta3zrOzfXspKVI+4+PdcS8vXpeEhBP1U1OmaNHRuQAAAAAs7e3BYCAgQB/8/OPIsbF57Vtbtr5qatRGy8uN2b6+Z0s5OXLeSkqU1ExMmOhYWLBKz8+Fa9DQuyrv78XlqqpPFvv77cVDQ4bXTU2aVTMzZpSFhRHPRUWKEPn56QYCAgSBf3/+8FBQoEQ8PHi6n58l46ioS/NRUaL+o6NdwEBAgIqPjwWtkpI/vJ2dIUg4OHAE9fXx37y8Y8G2tnd12tqvYyEhQjAQECAa///lDvPz/W3S0r9Mzc2BFAwMGDUTEyYv7OzD4V9fvqKXlzXMRESIORcXLlfExJPyp6dVgn5+/Ec9PXqsZGTI511duisZGTKVc3PmoGBgwJiBgRnRT0+ef9zco2YiIkR+KipUq5CQO4OIiAvKRkaMKe7ux9O4uGs8FBQoed7ep+JeXrwdCwsWdtvbrTvg4NtWMjJkTjo6dB4KChTbSUmSCgYGDGwkJEjkXFy4XcLCn27T073vrKxDpmJixKiRkTmklZUxN+Tk04t5efIy5+fVQ8jIi1k3N263bW3ajI2NAWTV1bHSTk6c4KmpSbRsbNj6VlasB/T08yXq6s+vZWXKjnp69OmurkcYCAgQ1bq6b4h4ePBvJSVKci4uXCQcHDjxpqZXx7S0c1HGxpcj6OjLfN3doZx0dOghHx8+3UtLlty9vWGGi4sNhYqKD5BwcOBCPj58xLW1capmZszYSEiQBQMDBgH29vcSDg4co2Fhwl81NWr5V1eu0Lm5aZGGhhdYwcGZJx0dOrmenic44eHZE/j467OYmCszEREiu2lp0nDZ2amJjo4Hp5SUM7abmy0iHh48koeHFSDp6clJzs6H/1VVqngoKFB639+lj4yMA/ihoVmAiYkJFw0NGtq/v2Ux5ubXxkJChLhoaNDDQUGCsJmZKXctLVoRDw8ey7Cwe/xUVKjWu7ttOhYWLA=='))
	T2 = array('I', b.b64decode('Y2PGpXx8+IR3d+6Ze3v2jfLy/w1ra9a9b2/escXFkVQwMGBQAQECA2dnzqkrK1Z9/v7nGdfXtWKrq03mdnbsmsrKj0WCgh+dycmJQH19+of6+u8VWVmy60dHjsnw8PsLra1B7NTUs2eiol/9r69F6pycI7+kpFP3cnLklsDAm1u3t3XC/f3hHJOTPa4mJkxqNjZsWj8/fkH39/UCzMyDTzQ0aFylpVH05eXRNPHx+QhxceKT2NirczExYlMVFSo/BAQIDMfHlVIjI0Zlw8OdXhgYMCiWljehBQUKD5qaL7UHBw4JEhIkNoCAG5vi4t896+vNJicnTmmysn/NdXXqnwkJEhuDgx2eLCxYdBoaNC4bGzYtbm7cslpatO6goFv7UlKk9js7dk3W1rdhs7N9zikpUnvj490+Ly9ecYSEE5dTU6b10dG5aAAAAADt7cEsICBAYPz84x+xsXnIW1u27Wpq1L7Ly41Gvr5n2Tk5cktKSpTeTEyY1FhYsOjPz4VK0NC7a+/vxSqqqk/l+/vtFkNDhsVNTZrXMzNmVYWFEZRFRYrP+fnpEAICBAZ/f/6BUFCg8Dw8eESfnyW6qKhL41FRovOjo13+QECAwI+PBYqSkj+tnZ0hvDg4cEj19fEEvLxj37a2d8Ha2q91ISFCYxAQIDD//+Ua8/P9DtLSv23NzYFMDAwYFBMTJjXs7MMvX1++4ZeXNaJERIjMFxcuOcTEk1enp1Xyfn78gj09ekdkZMisXV265xkZMitzc+aVYGDAoIGBGZhPT57R3NyjfyIiRGYqKlR+kJA7q4iIC4NGRozK7u7HKbi4a9MUFCg83t6neV5evOILCxYd29utduDg2zsyMmRWOjp0TgoKFB5JSZLbBgYMCiQkSGxcXLjkwsKfXdPTvW6srEPvYmLEppGROaiVlTGk5OTTN3l58ovn59UyyMiLQzc3blltbdq3jY0BjNXVsWROTpzSqalJ4Gxs2LRWVqz69PTzB+rqzyVlZcqvenr0jq6uR+kICBAYurpv1Xh48IglJUpvLi5cchwcOCSmplfxtLRzx8bGl1Ho6Msj3d2hfHR06JwfHz4hS0uW3b29YdyLiw2GiooPhXBw4JA+PnxCtbVxxGZmzKpISJDYAwMGBfb29wEODhwSYWHCozU1al9XV675ublp0IaGF5HBwZlYHR06J56eJ7nh4dk4+PjrE5iYK7MRESIzaWnSu9nZqXCOjgeJlJQzp5ubLbYeHjwih4cVkunpySDOzodJVVWq/ygoUHjf36V6jIwDj6GhWfiJiQmADQ0aF7+/Zdrm5tcxQkKExmho0LhBQYLDmZkpsC0tWncPDx4RsLB7y1RUqPy7u23WFhYsOg=='))
	T3 = array('I', b.b64decode('Y8alY3z4hHx37pl3e/aNe/L/DfJr1r1rb96xb8WRVMUwYFAwAQIDAWfOqWcrVn0r/ucZ/te1YterTearduyadsqPRcqCH52CyYlAyX36h3367xX6WbLrWUeOyUfw+wvwrUHsrdSzZ9SiX/2ir0Xqr5wjv5ykU/ekcuSWcsCbW8C3dcK3/eEc/ZM9rpMmTGomNmxaNj9+QT/39QL3zINPzDRoXDSlUfSl5dE05fH5CPFx4pNx2Ktz2DFiUzEVKj8VBAgMBMeVUscjRmUjw51ewxgwKBiWN6GWBQoPBZovtZoHDgkHEiQ2EoAbm4Di3z3i680m6ydOaSeyf82ydeqfdQkSGwmDHZ6DLFh0LBo0LhobNi0bbtyyblq07lqgW/ugUqT2Ujt2TTvWt2HWs33OsylSeynj3T7jL15xL4QTl4RTpvVT0blo0QAAAADtwSztIEBgIPzjH/yxecixW7btW2rUvmrLjUbLvmfZvjlySzlKlN5KTJjUTFiw6FjPhUrP0Ltr0O/FKu+qT+Wq++0W+0OGxUNNmtdNM2ZVM4URlIVFis9F+ekQ+QIEBgJ//oF/UKDwUDx4RDyfJbqfqEvjqFGi81GjXf6jQIDAQI8Fio+SP62SnSG8nThwSDj18QT1vGPfvLZ3wbbar3XaIUJjIRAgMBD/5Rr/8/0O89K/bdLNgUzNDBgUDBMmNRPswy/sX77hX5c1opdEiMxEFy45F8STV8SnVfKnfvyCfj16Rz1kyKxkXbrnXRkyKxlz5pVzYMCgYIEZmIFPntFP3KN/3CJEZiIqVH4qkDurkIgLg4hGjMpG7scp7rhr07gUKDwU3qd53l684l4LFh0L26122+DbO+AyZFYyOnROOgoUHgpJkttJBgwKBiRIbCRcuORcwp9dwtO9btOsQ++sYsSmYpE5qJGVMaSV5NM35Hnyi3nn1TLnyItDyDduWTdt2rdtjQGMjdWxZNVOnNJOqUngqWzYtGxWrPpW9PMH9OrPJeplyq9levSOeq5H6a4IEBgIum/VunjwiHglSm8lLlxyLhw4JBymV/GmtHPHtMaXUcboyyPo3aF83XTonHQfPiEfS5bdS71h3L2LDYaLig+FinDgkHA+fEI+tXHEtWbMqmZIkNhIAwYFA/b3AfYOHBIOYcKjYTVqXzVXrvlXuWnQuYYXkYbBmVjBHTonHZ4nuZ7h2Tjh+OsT+Jgrs5gRIjMRadK7admpcNmOB4mOlDOnlJsttpsePCIehxWSh+nJIOnOh0nOVar/VShQeCjfpXrfjAOPjKFZ+KGJCYCJDRoXDb9l2r/m1zHmQoTGQmjQuGhBgsNBmSmwmS1ady0PHhEPsHvLsFSo/FS7bda7Fiw6Fg=='))
	T4 = array('I', b.b64decode('xqVjY/iEfHzumXd39o17e/8N8vLWvWtr3rFvb5FUxcVgUDAwAgMBAc6pZ2dWfSsr5xn+/rVi19dN5qur7Jp2do9FysofnYKCiUDJyfqHfX3vFfr6sutZWY7JR0f7C/DwQeytrbNn1NRf/aKiReqvryO/nJxT96Sk5JZycptbwMB1wre34Rz9/T2uk5NMaiYmbFo2Nn5BPz/1Avf3g0/MzGhcNDRR9KWl0TTl5fkI8fHik3Fxq3PY2GJTMTEqPxUVCAwEBJVSx8dGZSMjnV7DwzAoGBg3oZaWCg8FBS+1mpoOCQcHJDYSEhubgIDfPeLizSbr605pJyd/zbKy6p91dRIbCQkdnoODWHQsLDQuGho2LRsb3LJubrTuWlpb+6CgpPZSUnZNOzu3YdbWfc6zs1J7KSndPuPjXnEvLxOXhISm9VNTuWjR0QAAAADBLO3tQGAgIOMf/Px5yLGxtu1bW9S+amqNRsvLZ9m+vnJLOTmU3kpKmNRMTLDoWFiFSs/Pu2vQ0MUq7+9P5aqq7Rb7+4bFQ0Oa101NZlUzMxGUhYWKz0VF6RD5+QQGAgL+gX9/oPBQUHhEPDwlup+fS+OoqKLzUVFd/qOjgMBAQAWKj48/rZKSIbydnXBIODjxBPX1Y9+8vHfBtravddraQmMhISAwEBDlGv///Q7z879t0tKBTM3NGBQMDCY1ExPDL+zsvuFfXzWil5eIzERELjkXF5NXxMRV8qen/IJ+fnpHPT3IrGRkuuddXTIrGRnmlXNzwKBgYBmYgYGe0U9Po3/c3ERmIiJUfioqO6uQkAuDiIiMykZGxynu7mvTuLgoPBQUp3ne3rziXl4WHQsLrXbb29s74OBkVjIydE46OhQeCgqS20lJDAoGBkhsJCS45Fxcn13Cwr1u09ND76ysxKZiYjmokZExpJWV0zfk5PKLeXnVMufni0PIyG5ZNzfat21tAYyNjbFk1dWc0k5OSeCpqdi0bGys+lZW8wf09M8l6urKr2Vl9I56ekfprq4QGAgIb9W6uvCIeHhKbyUlXHIuLjgkHBxX8aamc8e0tJdRxsbLI+jooXzd3eicdHQ+IR8flt1LS2Hcvb0NhouLD4WKiuCQcHB8Qj4+ccS1tcyqZmaQ2EhIBgUDA/cB9vYcEg4OwqNhYWpfNTWu+VdXadC5uReRhoaZWMHBOicdHSe5np7ZOOHh6xP4+CuzmJgiMxER0rtpaalw2dkHiY6OM6eUlC22m5s8Ih4eFZKHh8kg6emHSc7Oqv9VVVB4KCilet/fA4+MjFn4oaEJgImJGhcNDWXav7/XMebmhMZCQtC4aGiCw0FBKbCZmVp3LS0eEQ8Pe8uwsKj8VFRt1ru7LDoWFg=='))
	T5 = array('I', b.b64decode('UKf0UVNlQX7DpBcall4nOstrqzvxRZ0fq1j6rJMD40tV+jAg9m12rZF2zIglTAL1/NflT9fLKsWARDUmj6NitUlasd5nG7olmA7qReHA/l0CdS/DEvBMgaOXRo3G+dNr51+PA5WckhXrem2/2llSlS2DvtTTIXRYKWngSUTIyY5qicJ1eHmO9Gs+WJndcbkntk/hvhetiPBmrCDJtDrOfRhK32OCMRrlYDNRl0V/U2Lgd2SxhK5ruxyggf6UKwj5WGhIcBn9RY+HbN6Ut/h7UiPTc6viAktyV48f4yqrVWYHKOuyA8K1L5p7xYalCDfT8ocoMLKlvyO6agMCXIIW7Sscz4qStHmn8PIH86HiaU7N9Npl1b4FBh9iNNGK/qbEnVMuNKBV86Iy4YoFdev2pDnsgwuq72BABp9xXlEQbr35iiE+PQbdlq4FPt1GveZNtY1UkQVdxHFv1AYE/xVQYCT7mBmX6b3WzENAiXee2We9QuiwiIuJBzhbGefb7sh5Rwp8oekPQnzJHoT4AAAAAIOGgAlI7SsyrHARHk5yWmz7/w79VjiFDx7Vrj0nOS02ZNkPCiGmXGjRVFubOi42JLFnCgwP51eT0pbutJ6RmxtPxcCAoiDcYWlLd1oWGhIcCrqT4uUqoMBD4CI8HRcbEgsNCQ6tx4vyuai2LcipHhSFGfFXTAd1r7vdme79YH+jnyYB97z1clzFO2ZENH77W3YpQ4vcxiPLaPzttmPx5LjK3DHXEIVjQkAilxMgEcaEfSRKhfg9u9IRMvmubaEpx0svnh3zMLLc7FKGDdDjwXdsFrMrmblwqfpIlBEiZOlHxIz8qBo/8KDYLH1W75AzIsdOSYfB0TjZ/qLKjDYL1JjPgfWmKN56pSaOt9qkv60/5J06LA2SeFCbzF9qYkZ+VMITjfbouNiQXvc5LvWvw4K+gF2ffJPQaakt1W+zEiXPO5msyKd9GBBuY5zoe7s72wl4Js30GFluAbea7KiaT4NlbpXmfub/qgjPvCHm6BXv2Zvnus42b0rUCZ/q1nywKa+ypDExIz8qMJSlxsBmojU3vE50psqC/LDQkOAV2KczSpgE8ffa7EEOUM1/L/aRF43WTXZNsO9DVE2qzN8EluTjtdGeG4hqTLgfLMF/UWVGBOpenV01jAFzdIf6LkEL+1odZ7NS0tuSM1YQ6RNH1m2MYdeaegyhN44U+FmJPBPr7iepzjXJYbft5RzhPLFHelnf0pw/c/JVec4UGL83x3PqzfdTW6r9XxRvPd+G20R4gfOvyj7EaLksNCQ4X0CjwnLDHRYMJeK8i0k8KEGVDf9xAag53rMMCJzktNiQwVZkYYTLe3C2MtV0XGxIQle40A=='))
	T6 = array('I', b.b64decode('p/RRUGVBflOkFxrDXic6lmurO8tFnR/xWPqsqwPjS5P6MCBVbXat9nbMiJFMAvUl1+VP/MsqxddENSaAo2K1j1qx3kkbuiVnDupFmMD+XeF1L8MC8EyBEpdGjaP502vGX48D55ySFZV6bb/rWVKV2oO+1C0hdFjTaeBJKcjJjkSJwnVqeY70eD5YmWtxuSfdT+G+tq2I8BesIMlmOs59tErfYxgxGuWCM1GXYH9TYkV3ZLHgrmu7hKCB/hwrCPmUaEhwWP1Fjxls3pSH+HtSt9NzqyMCS3Lijx/jV6tVZioo67IHwrUvA3vFhpoIN9Olhygw8qW/I7JqAwK6ghbtXBzPiiu0eaeS8gfz8OJpTqH02mXNvgUG1WI00R/+psSKUy40nVXzoqDhigUy6/akdeyDCznvYECqn3FeBhBuvVGKIT75Bt2WPQU+3a695k1GjVSRtV3EcQXUBgRvFVBg//uYGSTpvdaXQ0CJzJ7ZZ3dC6LC9i4kHiFsZ5zjuyHnbCnyhRw9CfOkehPjJAAAAAIaACYPtKzJIcBEerHJabE7/Dv37OIUPVtWuPR45LTYn2Q8KZKZcaCFUW5vRLjYkOmcKDLHnV5MPlu600pGbG57FwIBPINxhokt3WmkaEhwWupPiCiqgwOXgIjxDFxsSHQ0JDgvHi/KtqLYtuakeFMgZ8VeFB3WvTN2Z7rtgf6P9JgH3n/VyXLw7ZkTFfvtbNClDi3bGI8vc/O22aPHkuGPcMdfKhWNCECKXE0ARxoQgJEqFfT270vgy+a4RoSnHbS+eHUswstzzUoYN7OPBd9AWsytsuXCpmUiUEfpk6UcijPyoxD/woBosfVbYkDMi705Jh8fRONnBosqM/gvUmDaB9abP3nqlKI632ia/rT+knTos5JJ4UA3MX2qbRn5UYhON9sK42JDo9zkuXq/DgvWAXZ++k9BpfC3Vb6kSJc+zmazIO30YEKdjnOhuuzvbe3gmzQkYWW70t5rsAZpPg6huleZl5v+qfs+8IQjoFe/mm+e62TZvSs4Jn+rUfLAp1rKkMa8jPyoxlKXGMGaiNcC8TnQ3yoL8ptCQ4LDYpzMVmATxStrsQfdQzX8O9pEXL9ZNdo2w70NNTarMVASW5N+10Z7jiGpMGx8swbhRZUZ/6l6dBDWMAV10h/pzQQv7Lh1ns1rS25JSVhDpM0fWbRNh15qMDKE3ehT4WY48E+uJJ6nO7slhtzXlHOHtsUd6PN/SnFlz8lU/zhQYeTfHc7/N91Pqqv1fW2893xTbRHiG86/KgcRouT40JDgsQKPCX8MdFnIl4rwMSTwoi5UN/0EBqDlxswwI3uS02JzBVmSQhMt7YbYy1XBcbEh0V7jQQg=='))
	T7 = array('I', b.b64decode('9FFQp0F+U2UXGsOkJzqWXqs7y2udH/FF+qyrWONLkwMwIFX6dq32bcyIkXYC9SVM5U/81yrF18s1JoBEYrWPo7HeSVq6JWcb6kWYDv5d4cAvwwJ1TIES8EaNo5fTa8b5jwPnX5IVlZxtv+t6UpXaWb7ULYN0WNMh4EkpacmORMjCdWqJjvR4eViZaz65J91x4b62T4jwF60gyWaszn20Ot9jGEoa5YIxUZdgM1NiRX9kseB3a7uEroH+HKAI+ZQrSHBYaEWPGf3elIdse1K3+HOrI9NLcuICH+NXj1VmKqvrsgcotS8DwsWGmns306UIKDDyh78jsqUDArpqFu1cgs+KKxx5p5K0B/Pw8mlOoeLaZc30BQbVvjTRH2KmxIr+LjSdU/OioFWKBTLh9qR164MLOexgQKrvcV4Gn269URAhPvmK3ZY9Bj7drgXmTUa9VJG1jcRxBV0GBG/UUGD/FZgZJPu91pfpQInMQ9lnd57osL1CiQeIixnnOFvIedvufKFHCkJ86Q+E+MkeAAAAAIAJg4YrMkjtER6scFpsTnIO/fv/hQ9WOK49HtUtNic5Dwpk2VxoIaZbm9FUNiQ6LgoMsWdXkw/n7rTSlpsbnpHAgE/F3GGiIHdaaUsSHBYak+IKuqDA5SoiPEPgGxIdFwkOCw2L8q3Hti25qB4UyKnxV4UZda9MB5nuu91/o/1gAfefJnJcvPVmRMU7+1s0fkOLdikjy9zG7bZo/OS4Y/Ex18rcY0IQhZcTQCLGhCARSoV9JLvS+D35rhEyKcdtoZ4dSy+y3PMwhg3sUsF30OOzK2wWcKmZuZQR+kjpRyJk/KjEjPCgGj99VtgsMyLvkEmHx0442cHRyoz+otSYNgv1ps+BeqUo3rfaJo6tP6S/OizknXhQDZJfapvMflRiRo32whPYkOi4OS5e98OC9a9dn76A0Gl8k9VvqS0lz7MSrMg7mRgQp32c6G5jO9t7uybNCXhZbvQYmuwBt0+DqJqV5mVu/6p+5rwhCM8V7+bo57rZm29Kzjaf6tQJsCnWfKQxr7I/KjEjpcYwlKI1wGZOdDe8gvymypDgsNCnMxXYBPFKmOxB99rNfw5QkRcv9k12jdbvQ02wqsxUTZbk3wTRnuO1akwbiCzBuB9lRn9RXp0E6owBXTWH+nN0C/suQWezWh3bklLSEOkzVtZtE0fXmoxhoTd6DPhZjhQT64k8qc7uJ2G3Nckc4e3lR3o8sdKcWd/yVT9zFBh5zsdzvzf3U+rN/V9bqj3fFG9EeIbbr8qB82i5PsQkOCw0o8JfQB0WcsPivAwlPCiLSQ3/QZWoOXEBDAjes7TYnORWZJDBy3thhDLVcLZsSHRcuNBCVw=='))
	T8 = array('I', b.b64decode('UVCn9H5TZUEaw6QXOpZeJzvLa6sf8UWdrKtY+kuTA+MgVfowrfZtdoiRdsz1JUwCT/zX5cXXyyomgEQ1tY+jYt5JWrElZxu6RZgO6l3hwP7DAnUvgRLwTI2jl0ZrxvnTA+dfjxWVnJK/63ptldpZUtQtg75Y0yF0SSlp4I5EyMl1aonC9Hh5jplrPlgn3XG5vrZP4fAXrYjJZqwgfbQ6zmMYSt/lgjEal2AzUWJFf1Ox4Hdku4Sua/4coIH5lCsIcFhoSI8Z/UWUh2zeUrf4e6sj03Ny4gJL41ePH2Yqq1WyByjrLwPCtYaae8XTpQg3MPKHKCOypb8CumoD7VyCFoorHM+nkrR58/DyB06h4mllzfTaBtW+BdEfYjTEiv6mNJ1TLqKgVfMFMuGKpHXr9gs57INAqu9gXgafcb1REG4++Yohlj0G3d2uBT5NRr3mkbWNVHEFXcQEb9QGYP8VUBkk+5jWl+m9icxDQGd3ntmwvULoB4iLiec4Wxl52+7IoUcKfHzpD0L4yR6EAAAAAAmDhoAySO0rHqxwEWxOclr9+/8OD1Y4hT0e1a42JzktCmTZD2ghplyb0VRbJDouNgyxZwqTD+dXtNKW7huekZuAT8XAYaIg3FppS3ccFhoS4gq6k8DlKqA8Q+AiEh0XGw4LDQnyrceLLbmothTIqR5XhRnxr0wHde673Zmj/WB/958mAVy89XJExTtmWzR++4t2KUPL3MYjtmj87bhj8eTXytwxQhCFYxNAIpeEIBHGhX0kStL4PbuuETL5x22hKR1LL57c8zCyDexShnfQ48ErbBazqZm5cBH6SJRHImTpqMSM/KAaP/BW2Cx9Iu+QM4fHTknZwdE4jP6iypg2C9Smz4H1pSjeetomjrc/pL+tLOSdOlANknhqm8xfVGJGfvbCE42Q6LjYLl73OYL1r8OfvoBdaXyT0G+pLdXPsxIlyDuZrBCnfRjobmOc23u7O80JeCZu9BhZ7AG3moOomk/mZW6Vqn7m/yEIz7zv5ugVutmb50rONm/q1AmfKdZ8sDGvsqQqMSM/xjCUpTXAZqJ0N7xO/KbKguCw0JAzFdin8UqYBEH32ux/DlDNFy/2kXaN1k1DTbDvzFRNquTfBJae47XRTBuIasG4HyxGf1FlnQTqXgFdNYz6c3SH+y5BC7NaHWeSUtLb6TNWEG0TR9aajGHXN3oMoVmOFPjriTwTzu4nqbc1yWHh7eUcejyxR5xZ39JVP3PyGHnOFHO/N8dT6s33X1uq/d8Ubz14httEyoHzr7k+xGg4LDQkwl9AoxZywx28DCXiKItJPP9BlQ05cQGoCN6zDNic5LRkkMFWe2GEy9VwtjJIdFxs0EJXuA=='))
	U1 = array('I', b.b64decode('AAAAAAsNCQ4WGhIcHRcbEiw0JDgnOS02Oi42JDEjPypYaEhwU2VBfk5yWmxFf1NidFxsSH9RZUZiRn5UaUt3WrDQkOC73ZnupsqC/K3Hi/Kc5LTYl+m91or+psSB86/K6LjYkOO10Z7+osqM9a/DgsSM/KjPgfWm0pbutNmb57p7uzvbcLYy1W2hKcdmrCDJV48f41yCFu1BlQ3/SpgE8SPTc6so3nqlNclhtz7EaLkP51eTBOpenRn9RY8S8EyBy2urO8BmojXdcbkn1nywKedfjwPsUoYN8UWdH/pIlBGTA+NLmA7qRYUZ8VeOFPhZvzfHc7Q6zn2pLdVvoiDcYfZtdq39YH+j4Hdkset6bb/aWVKV0VRbm8xDQInHTkmHrgU+3aUIN9O4HyzBsxIlz4IxGuWJPBPrlCsI+Z8mAfdGveZNTbDvQ1Cn9FFbqv1faonCdWGEy3t8k9Bpd57ZZx7Vrj0V2KczCM+8IQPCtS8y4YoFOeyDCyT7mBkv9pEXjdZNdobbRHibzF9qkMFWZKHiaU6q72BAt/h7Urz1clzVvgUG3rMMCMOkFxrIqR4U+YohPvKHKDDvkDMi5J06LD0G3ZY2C9SYKxzPiiARxoQRMvmuGj/woAco67IMJeK8ZW6V5m5jnOhzdIf6eHmO9Elasd5CV7jQX0CjwlRNqsz32uxB/NflT+HA/l3qzfdT2+7IedDjwXfN9NplxvnTa6+ypDGkv60/uai2LbKlvyODhoAJiIuJB5WckhWekZsbRwp8oUwHda9REG69Wh1ns2s+WJlgM1GXfSRKhXYpQ4sfYjTRFG893wl4Js0CdS/DM1YQ6ThbGeclTAL1LkEL+4xh15qHbN6UmnvFhpF2zIigVfOiq1j6rLZP4b69Quiw1Amf6t8EluTCE432yR6E+Pg9u9LzMLLc7iepzuUqoMA8sUd6N7xOdCqrVWYhplxoEIVjQhuIakwGn3FeDZJ4UGTZDwpv1AYEcsMdFnnOFBhI7SsyQ+AiPF73OS5V+jAgAbea7Aq6k+IXrYjwHKCB/i2DvtQmjrfaO5msyDCUpcZZ39KcUtLbkk/FwIBEyMmOdev2pH7m/6pj8eS4aPzttrFnCgy6agMCp30YEKxwER6dUy40ll4nOotJPCiARDUm6Q9CfOICS3L/FVBg9BhZbsU7ZkTONm9K0yF0WNgsfVZ6DKE3cQGoOWwWsytnG7olVjiFD101jAFAIpcTSy+eHSJk6UcpaeBJNH77Wz9z8lUOUM1/BV3EcRhK32MTR9Ztytwx18HRONncxiPL18sqxeboFe/t5Rzh8PIH8/v/Dv2StHmnmblwqYSua7uPo2K1voBdn7WNVJGomk+Do5dGjQ=='))
	U2 = array('I', b.b64decode('AAAAAA0JDgsaEhwWFxsSHTQkOCw5LTYnLjYkOiM/KjFoSHBYZUF+U3JabE5/U2JFXGxIdFFlRn9GflRiS3daadCQ4LDdme67yoL8pseL8q3ktNic6b3Wl/6mxIrzr8qBuNiQ6LXRnuOiyoz+r8OC9Yz8qMSB9abPlu600pvnutm7O9t7tjLVcKEpx22sIMlmjx/jV4IW7VyVDf9BmATxStNzqyPeeqUoyWG3NcRouT7nV5MP6l6dBP1FjxnwTIESa6s7y2aiNcBxuSfdfLAp1l+PA+dShg3sRZ0f8UiUEfoD40uTDupFmBnxV4UU+FmON8dzvzrOfbQt1W+pINxhom12rfZgf6P9d2Sx4Hptv+tZUpXaVFub0UNAicxOSYfHBT7drgg306UfLMG4EiXPszEa5YI8E+uJKwj5lCYB95+95k1GsO9DTaf0UVCq/V9bicJ1aoTLe2GT0Gl8ntlnd9WuPR7YpzMVz7whCMK1LwPhigUy7IMLOfuYGST2kRcv1k12jdtEeIbMX2qbwVZkkOJpTqHvYECq+HtSt/VyXLy+BQbVswwI3qQXGsOpHhTIiiE++YcoMPKQMyLvnTos5Abdlj0L1Jg2HM+KKxHGhCAy+a4RP/CgGijrsgcl4rwMbpXmZWOc6G50h/pzeY70eFqx3klXuNBCQKPCX02qzFTa7EH31+VP/MD+XeHN91Pq7sh52+PBd9D02mXN+dNrxrKkMa+/rT+kqLYtuaW/I7KGgAmDi4kHiJySFZWRmxueCnyhRwd1r0wQbr1RHWezWj5YmWszUZdgJEqFfSlDi3ZiNNEfbz3fFHgmzQl1L8MCVhDpM1sZ5zhMAvUlQQv7LmHXmoxs3pSHe8WGmnbMiJFV86KgWPqsq0/hvrZC6LC9CZ/q1ASW5N8TjfbCHoT4yT270vgwstzzJ6nO7iqgwOWxR3o8vE50N6tVZiqmXGghhWNCEIhqTBufcV4GknhQDdkPCmTUBgRvwx0Wcs4UGHntKzJI4CI8Q/c5Ll76MCBVt5rsAbqT4gqtiPAXoIH+HIO+1C2Ot9ommazIO5SlxjDf0pxZ0tuSUsXAgE/IyY5E6/akdeb/qn7x5Lhj/O22aGcKDLFqAwK6fRgQp3ARHqxTLjSdXic6lkk8KItENSaAD0J86QJLcuIVUGD/GFlu9DtmRMU2b0rOIXRY0yx9VtgMoTd6Aag5cRazK2wbuiVnOIUPVjWMAV0ilxNAL54dS2TpRyJp4EkpfvtbNHPyVT9QzX8OXcRxBUrfYxhH1m0T3DHXytE42cHGI8vcyyrF1+gV7+blHOHt8gfz8P8O/fu0eaeSuXCpma5ru4SjYrWPgF2fvo1UkbWaT4Ool0aNow=='))
	U3 = array('I', b.b64decode('AAAAAAkOCw0SHBYaGxIdFyQ4LDQtNic5NiQ6Lj8qMSNIcFhoQX5TZVpsTnJTYkV/bEh0XGVGf1F+VGJGd1ppS5DgsNCZ7rvdgvymyovyrce02JzkvdaX6abEiv6vyoHz2JDouNGe47XKjP6iw4L1r/yoxIz1ps+B7rTSlue62Zs723u7MtVwtinHbaEgyWasH+NXjxbtXIIN/0GVBPFKmHOrI9N6pSjeYbc1yWi5PsRXkw/nXp0E6kWPGf1MgRLwqzvLa6I1wGa5J91xsCnWfI8D51+GDexSnR/xRZQR+kjjS5MD6kWYDvFXhRn4WY4Ux3O/N859tDrVb6kt3GGiIHat9m1/o/1gZLHgd22/63pSldpZW5vRVECJzENJh8dOPt2uBTfTpQgswbgfJc+zEhrlgjET64k8CPmUKwH3nybmTUa970NNsPRRUKf9X1uqwnVqict7YYTQaXyT2Wd3nq49HtWnMxXYvCEIz7UvA8KKBTLhgws57JgZJPuRFy/2TXaN1kR4httfapvMVmSQwWlOoeJgQKrve1K3+HJcvPUFBtW+DAjesxcaw6QeFMipIT75iigw8oczIu+QOizknd2WPQbUmDYLz4orHMaEIBH5rhEy8KAaP+uyByjivAwlleZlbpzobmOH+nN0jvR4ebHeSVq40EJXo8JfQKrMVE3sQffa5U/81/5d4cD3U+rNyHnb7sF30OPaZc3002vG+aQxr7KtP6S/ti25qL8jsqWACYOGiQeIi5IVlZybG56RfKFHCnWvTAduvVEQZ7NaHViZaz5Rl2AzSoV9JEOLdik00R9iPd8UbybNCXgvwwJ1EOkzVhnnOFsC9SVMC/suQdeajGHelIdsxYaae8yIkXbzoqBV+qyrWOG+tk/osL1Cn+rUCZbk3wSN9sIThPjJHrvS+D2y3PMwqc7uJ6DA5SpHejyxTnQ3vFVmKqtcaCGmY0IQhWpMG4hxXgafeFANkg8KZNkGBG/UHRZywxQYec4rMkjtIjxD4DkuXvcwIFX6muwBt5PiCrqI8Betgf4coL7ULYO32iaOrMg7maXGMJTSnFnf25JS0sCAT8XJjkTI9qR16/+qfubkuGPx7bZo/AoMsWcDArpqGBCnfREerHAuNJ1TJzqWXjwoi0k1JoBEQnzpD0ty4gJQYP8VWW70GGZExTtvSs42dFjTIX1W2CyhN3oMqDlxAbMrbBa6JWcbhQ9WOIwBXTWXE0Ainh1LL+lHImTgSSlp+1s0fvJVP3PNfw5QxHEFXd9jGErWbRNHMdfK3DjZwdEjy9zGKsXXyxXv5ugc4e3lB/Pw8g79+/95p5K0cKmZuWu7hK5itY+jXZ++gFSRtY1Pg6iaRo2jlw=='))
	U4 = array('I', b.b64decode('AAAAAA4LDQkcFhoSEh0XGzgsNCQ2JzktJDouNioxIz9wWGhIflNlQWxOclpiRX9TSHRcbEZ/UWVUYkZ+WmlLd+Cw0JDuu92Z/KbKgvKtx4vYnOS01pfpvcSK/qbKgfOvkOi42J7jtdGM/qLKgvWvw6jEjPymz4H1tNKW7rrZm+fbe7s71XC2MsdtoSnJZqwg41ePH+1cghb/QZUN8UqYBKsj03OlKN56tzXJYbk+xGiTD+dXnQTqXo8Z/UWBEvBMO8trqzXAZqIn3XG5KdZ8sAPnX48N7FKGH/FFnRH6SJRLkwPjRZgO6leFGfFZjhT4c783x320Os5vqS3VYaIg3K32bXaj/WB/seB3ZL/rem2V2llSm9FUW4nMQ0CHx05J3a4FPtOlCDfBuB8sz7MSJeWCMRrriTwT+ZQrCPefJgFNRr3mQ02w71FQp/RfW6r9dWqJwnthhMtpfJPQZ3ee2T0e1a4zFdinIQjPvC8DwrUFMuGKCznsgxkk+5gXL/aRdo3WTXiG20Rqm8xfZJDBVk6h4mlAqu9gUrf4e1y89XIG1b4FCN6zDBrDpBcUyKkePvmKITDyhygi75AzLOSdOpY9Bt2YNgvUiiscz4QgEcauETL5oBo/8LIHKOu8DCXi5mVulehuY5z6c3SH9Hh5jt5JWrHQQle4wl9Ao8xUTapB99rsT/zX5V3hwP5T6s33edvuyHfQ48FlzfTaa8b50zGvsqQ/pL+tLbmotiOypb8Jg4aAB4iLiRWVnJIbnpGboUcKfK9MB3W9URBus1odZ5lrPliXYDNRhX0kSot2KUPRH2I03xRvPc0JeCbDAnUv6TNWEOc4Wxn1JUwC+y5BC5qMYdeUh2zehpp7xYiRdsyioFXzrKtY+r62T+GwvULo6tQJn+TfBJb2whON+MkehNL4Pbvc8zCyzu4nqcDlKqB6PLFHdDe8TmYqq1VoIaZcQhCFY0wbiGpeBp9xUA2SeApk2Q8Eb9QGFnLDHRh5zhQySO0rPEPgIi5e9zkgVfow7AG3muIKupPwF62I/hyggdQtg77aJo63yDuZrMYwlKWcWd/SklLS24BPxcCORMjJpHXr9qp+5v+4Y/Hktmj87QyxZwoCumoDEKd9GB6scBE0nVMuOpZeJyiLSTwmgEQ1fOkPQnLiAktg/xVQbvQYWUTFO2ZKzjZvWNMhdFbYLH03egyhOXEBqCtsFrMlZxu6D1Y4hQFdNYwTQCKXHUsvnkciZOlJKWngWzR++1U/c/J/DlDNcQVdxGMYSt9tE0fW18rcMdnB0TjL3MYjxdfLKu/m6BXh7eUc8/DyB/37/w6nkrR5qZm5cLuErmu1j6Nin76AXZG1jVSDqJpPjaOXRg=='))
```

Looks like it would reduce the size of this chunk by about 16,000 bytes (~48%). When I profiled it on Python3, it was *slightly* slower which was surprising, I thought it'd be faster. If speed is a concern, converting each array to a tuple got the speed back to where it is here.

Reply:
If this is something that only gets done once, maybe the speed isn't too much of a concern. Is the binary data ABI-compatible between python versions?

Reply:
Yup, the `array` module goes back to at least Python 2.5.

Reply:
I had indeed pondered this too. I'm glad you lads brought it up. They're pretty gross looking inline. We could do the gzip thing with these too?

Reply:
@smcintyre-r7 I think the up-front cost is bearable given that we are only doing it once. Optimising for size is probably the best option.